packages/picasso/src/TimePicker/TimePicker.tsx

```diff
@@ -58,6 +60,25 @@ export const TimePicker = (props: Props) => {
     ...rest
   } = props
 
+  const [value, setValue] = useState(initialValue)
+
+  const onChange = (
+    event: React.ChangeEvent<
+      HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+    >
+  ) => {
+    const newValue = event.target.value
+
+    setValue(newValue)
+
+    if (newValue && VALID_TIME_REGEX.test(newValue)) {
+      externalOnChange?.(event)
+    } else {
+      event.target.value = ''
+      externalOnChange?.(event)
+    }
```

Comment:
This seems a bit counter-intuitive, we are intentionally keeping `value` out of sync with the element's value prop. This goes against the concept of [controlled components](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable), and might create hard to debug issues.

I would prefer if instead of changing `event.target.value` directly we would keep a state variable to it, and just update it.
```suggestion
    setValue(newValue)

    if (newValue && VALID_TIME_REGEX.test(newValue)) {
      externalOnChange?.(event)
    } else {
      // Just an example, not sure if it would work with the remaining code
      setState('')
      externalOnChange?.(event)
    }
```